orbit/pkg/update/flag_runner.go

```diff
@@ -188,3 +196,22 @@ func readFlagFile(rootDir string) (map[string]string, error) {
 	}
 	return result, nil
 }
+
+// updateOrbitNodeKey does re-enrolls by calling the /enroll API and writes the response to disk
+func (r *FlagRunner) updateOrbitNodeKey() (string, error) {
```

Comment:
Should this instead be part of `service.OrbitClient`?

Reply:
I did start with that, but given that `FlagUpdateOptions` struct in `FlagRunner` has `OrbitNodeKey` as a field, which we need to update, and also write to disk, I thought this might be a better place

Reply:
Mhm... `OrbitNodeKey` should be part of the `service.OrbitClient`.

Reply:
If this works, given the urgency of the release, am ok with merging, but we need to refactor this as soon as we add more functionality to Orbit.

Reply:
`enrollSecret` is part of `service.OrbitClient`, given it only needs that along with `hardwareUUID` to make an `/enroll` request

Reply:
(Not needed for this PR) But if I were to use a `service.OrbitClient` I would expect it would handle all the logic to re-enroll on auth failure, handle/load its orbit node key, etc.

Reply:
yep, this works and makes sense to refactor, planning on incorporating it in extension autoupdat

Reply:
when we refactor, we could use pkg/retry for the retries too :) https://github.com/fleetdm/fleet/blob/main/pkg/retry/retry.go

Reply:
cool!